tests/contest/contest/src/tests/net_devices/net_devices_test.rs

```diff
@@ -0,0 +1,389 @@
+            "netns",
+            "1",
+        ])
+        .output()
+    {
+        return TestResult::Failed(anyhow!("Failed to move device back: {}", e));
+    }
+
+    // Check that the device exists
+    if let Err(e) = check_device_exists(&device_name) {
+        return TestResult::Failed(anyhow!("Failed to check device: {}", e));
+    }
+
+    if let Err(e) = delete_dummy_device(&device_name) {
+        return TestResult::Failed(anyhow!("Failed to delete device: {}", e));
+    }
+
+    if let Err(e) = cleanup_netns(&netns_name) {
+        return TestResult::Failed(anyhow!("Failed to cleanup netns: {}", e));
+    }
+
+    TestResult::Passed
+}
+
+fn check_address() -> TestResult {
+    let netns_name = create_unique_netns_name("netns-address");
+    let device_name = create_unique_device_name("dummy-address");
+    const DUMMY_ADDRESS: &str = "244.178.44.111/24";
+
+    let mut net_devices = HashMap::new();
+    net_devices.insert(
+        device_name.clone(),
+        LinuxNetDeviceBuilder::default()
+            .name(&device_name)
+            .build()
+            .unwrap(),
+    );
+
+    if let Err(e) = create_netns(&netns_name) {
+        return TestResult::Failed(anyhow!("Failed to create netns: {}", e));
+    }
+
+    if let Err(e) = create_dummy_device(&device_name) {
+        return TestResult::Failed(anyhow!("Failed to create dummy device: {}", e));
+    }
+
+    // Add address to the device
+    if let Err(e) = std::process::Command::new("ip")
+        .args(vec!["addr", "add", DUMMY_ADDRESS, "dev", &device_name])
+        .output()
+    {
+        return TestResult::Failed(anyhow!("Failed to add address: {}", e));
+    }
+
+    let spec = create_spec_with_netns(
+        net_devices,
+        Path::new("/var/run/netns")
+            .join(&netns_name)
+            .to_str()
+            .unwrap()
+            .to_string(),
+    );
+    let test_result = test_outside_container(&spec, &|data| {
+        test_result!(check_container_created(&data));
+        TestResult::Passed
+    });
+    if let TestResult::Failed(_) = test_result {
+        return test_result;
+    }
+
+    // Check that the address was added
+    let out = match std::process::Command::new("ip")
+        .args(vec!["netns", "exec", &netns_name, "ip", "addr"])
+        .output()
+    {
+        Ok(out) => out,
+        Err(e) => return TestResult::Failed(anyhow!("Failed to check address: {}", e)),
+    };
+    let out = match String::from_utf8(out.stdout) {
+        Ok(out) => out,
+        Err(e) => return TestResult::Failed(anyhow!("Failed to parse output: {}", e)),
+    };
+    if !out.contains(&device_name) || !out.contains(DUMMY_ADDRESS) {
+        return TestResult::Failed(anyhow!("Address not found in output"));
+    }
+
+    if let Err(e) = cleanup_netns(&netns_name) {
+        return TestResult::Failed(anyhow!("Failed to cleanup netns: {}", e));
+    }
+
+    TestResult::Passed
+}
+
+pub fn get_net_devices_test() -> TestGroup {
```

Comment:
If you want to skip tests when running with runc, you can use the `is_runtime_runc` function together with  `ConditionalTest`.

Please refer to:
https://github.com/youki-dev/youki/blob/main/tests/contest/contest/src/tests/fd_control/mod.rs#L94